### PR TITLE
Eff cov2: Union effect support + bug fixes

### PR DESCRIPTION
**Pull request by bhavik-goplani**

### CODE REVIEW

File: test/test_effects.rb [hunk 1]
```diff
@@ -7,12 +7,12 @@
 class TestEffects < Minitest::Test
   extend RDL::Annotate
 
-  type :pure_method, '(Integer) -> Integer [.]', typecheck: :effects
-  def pure_method(x)
-    x
-  end
+  # type :pure_method, '(Integer) -> Integer [.]', typecheck: :effects
```
Comment (ngsankha) [inline test/test_effects.rb]: Uncomment this test

**Review by ngsankha — CHANGES_REQUESTED** [review]
Overall looks good to me. Left one inline comment to fix. All the extra `puts` pollutes the console, it would be better to remove these. It is fine for now, we can revisit that and remove that later.

File: test/test_effects.rb (hunk 1 above)
Comment (bhavik-goplani) [reply test/test_effects.rb]: 2c975cf07cadfb7fa87e4e81f2a45803d02d9222
Uncommented the test.

**Comment by bhavik-goplani** [comment]
> Overall looks good to me. Left one inline comment to fix. All the extra `puts` pollutes the console, it would be better to remove these. It is fine for now, we can revisit that and remove that later.

Thanks for the feedback. We can remove those puts once I test this on the issues.

**Pull request merged**